pennylane/gradients/pulse_gradient.py

```diff
@@ -188,9 +225,13 @@ def _stoch_pulse_grad(tape, argnum=None, num_split_times=1, sampler_seed=None, s
             rule underlying the differentiation; also see details
         sample_seed (int): randomness seed to be used for the time samples in the stochastic
             parameter-shift rule
-        shots (None, int, list[int]): The device shots that will be used to execute the tapes outputted by this
-            transform. Note that this argument doesn't influence the shots used for tape execution, but provides
-            information about the shots.
+        shots (None, int, list[int]): The device shots that will be used to execute the tapes
+            outputted by this transform. Note that this argument doesn't influence the shots
+            used for tape execution, but provides information about the shots.
+        use_broadcasting (bool): Whether to use broadcasting across the different sampled
+            splitting times. If ``False`` (the default), one set of modified tapes per
+            splitting time is created, if ``True`` only a single set of broadcasted, modified
+            tapes is created.
```

Comment:
Makes sense to mention that this option is good for simulators and speeds things up 👍 I guess for hardware this never makes sense / works right?

Reply:
Yep, added a comment regarding this :)

Reply:
small amendment https://github.com/PennyLaneAI/pennylane/pull/4004/files#r1169822846